src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -121,13 +248,33 @@ public bool UseDefaultCredentials
             return proxyUri;
         }
 
-        private void UpdateRegexList(bool canThrow)
+
+        //Replaces calls to UpdateRegexList wich took a 'canThrow' flag, set to false, so we just swallow any exception.
+        private void UpdateRegexListIfBypassListIsDirty()
+        {
+            DirtyTrackingArrayList? bypassList = _bypassList;
+            if (bypassList != null && bypassList.CheckDirty())
+            {
+                try
+                {
+                    UpdateRegexList();
+                }
+                catch
+                {
+                    _regexBypassList = null;
+                }
+            }
+        }
+
+        private void UpdateRegexList()
         {
+            DirtyTrackingArrayList? bypassList = _bypassList;
+
             Regex[]? regexBypassList = null;
-            ArrayList? bypassList = _bypassList;
-            try
+            if (bypassList != null)
             {
-                if (bypassList != null && bypassList.Count > 0)
+                bypassList.CheckDirty();
```

Comment:
What's the purpose of calling CheckDirty if the return value is ignored?  If the goal here is to just set IsDirty to false, it'd be better to do that explicitly.  And at that point, we don't need CheckDirty... the other call site can similarly just access the getter, and then if it's true and we enter that if block, set IsDirty to false.

Reply:
Yes, CheckDirty is not really a good name.

The initial idea was to avoid having to do a 
if (a.IsDirty) {
 // do some stuff
 a.IsDirty = false;
}

and rather simply do

if (a.CheckDirty()) {
  // do some stuff
}
Plus I though the dirty flag should only be set to true by the class, not by an external caller, so I didn't want to expose a setter for IsDirty.

But turns out, that's not it's not really used that way...


And yes, the reason it's called in the UpdateRegexList is to reset the dirty status to false for when the UpdateRegexList is called directly in the constructor and the BypassList setter. In that case, we don't care about the return value.

But if it's clearer, we can do

IsChanged {get;}
and void ResetIsChanged();

Reply:
Right. And at that point it can just be an IsChanged with a getter and setter.

Reply:
I committed a second set of changes based on your general feedback.
But I did IsChanged  {get;} / void ResetIsChanged()
I'm still not too fond of beeing able to do a.IsChanged = true; from an external caller.

Reply:
> I'm still not too fond of beeing able to do a.IsChanged = true; from an external caller. (even if this this all completely private stuff)

Why?  `ResetIsChanged()` is equivalent to setting `IsChanged` to false, and there are a myriad of other ways to set it to true, e.g. `a[0] = a[0];`, so not exposing `IsChanged` because of concerns around someone doing `IsChanged = true;` isn't preventing anything.  Honestly my inclination would be that a property is even too much here and it should just be an internal `_isChanged` field.  This type isn't even available to the rest of the assembly, nevermind public... it's a private implementation detail of WebProxy, and in that regard is just like any other private member, whose shape is there purely to serve the needs of the type it's private in.  We shouldn't spend time or energy on polishing the shape of these APIs beyond what's helpful in maintainability.  And FWIW, from my perspective the most maintainable thing is for there to be a single entity that's used to both get and set the flag.